kie-plugins-testing/src/test/java/org/kie/maven/plugin/CompilerHelperTest.java

```diff
@@ -0,0 +1,518 @@
+    }
+
+    @Test
+    public void getWrongCompilationIDTest() {
+        CompilerHelper helper = new CompilerHelper();
+        Map<String, Object> map = new HashMap<>();
+        String uuid = UUID.randomUUID().toString();
+        map.put("compilation..ID",
+                uuid);
+        String compilationID = helper.getCompilationID(map,
+                                                       log);
+        assertThat(compilationID).isEqualTo("main");
+    }
+
+    @Test
+    public void shareStoreWithMapTest() {
+        //prepare
+        CompilerHelper helper = new CompilerHelper();
+        Map<String, Object> kieMap = new HashMap<>();
+        String compilationID = UUID.randomUUID().toString();
+        kieMap.put("compilation.ID",
+                   compilationID);
+        ProjectClassLoader pcl = ProjectClassLoader.createProjectClassLoader();
+        String hexContent = "e04fd020ea3a6910a2d808002b30309d";
+        String className = "org/my/Clazz";
+        //we store bytes of hexString for an easy compare
+        pcl.storeClass(className,
+                       hexStringToByteArray(hexContent));
+
+        //call the class under test
+        helper.shareStoreWithMap(pcl,
+                                 compilationID,
+                                 kieMap,
+                                 log);
+
+        //verify
+        StringBuilder sbTypes = new StringBuilder(compilationID).append(".").append("ProjectClassloaderStore");
+        kieMap.get(sbTypes.toString());
+        Map store = (Map) kieMap.get(sbTypes.toString());
+        assertThat(store).isNotNull();
+        assertThat(store.size()).isEqualTo(1);
+        byte[] bytez = (byte[]) store.get(className + ".class");
+        assertThat(bytez).isNotNull();
+        String hex = toHexString(bytez);
+        assertThat(hex).isEqualTo(hexContent);
+    }
+
+    @Test
+    public void shareNoStoreWithMapTest() {
+        //prepare
+        CompilerHelper helper = new CompilerHelper();
+        Map<String, Object> kieMap = new HashMap<>();
+        String compilationID = UUID.randomUUID().toString();
+        kieMap.put("compilation.ID",
+                   compilationID);
+        ProjectClassLoader pcl = ProjectClassLoader.createProjectClassLoader();
+
+        //call the class under test
+        helper.shareStoreWithMap(pcl,
+                                 compilationID,
+                                 kieMap,
+                                 log);
+
+        //verify
+        StringBuilder sbTypes = new StringBuilder(compilationID).append(".").append("ProjectClassloaderStore");
+        kieMap.get(sbTypes.toString());
+        Map store = (Map) kieMap.get(sbTypes.toString());
+        assertThat(store).isNull();
+    }
+
+    @Test
+    public void shareWrongStoreWithMapTest() throws Exception {
+        //prepare
+        CompilerHelper helper = new CompilerHelper();
+        Map<String, Object> kieMap = new HashMap<>();
```

Comment:
Just so you know: you can create a map with single key-value pair using single statement
```
Map<String,Object> kieMap = Collections.singletonMap("compilation.ID", compilationID);
``` 
to remove some boilerplate..